addon/utils/global-text-region-to-model-range.ts

```diff
@@ -0,0 +1,84 @@
+import ModelNode from "@lblod/ember-rdfa-editor/model/model-node";
+import ModelTreeWalker, { FilterResult } from "@lblod/ember-rdfa-editor/model/util/model-tree-walker";
+import ModelElement from "@lblod/ember-rdfa-editor/model/model-element";
+import ModelRange from "@lblod/ember-rdfa-editor/model/model-range";
+import ModelText from "../model/model-text";
+import ModelPosition from "../model/model-position";
+
+const VOID_NODES = ["area", "base", "br", "col", "command", "embed", "hr", "img", "input", "keygen", "link", "meta", "param", "source", "track", "wbr"];
+
+type ModelTextWithOffsets = { node: ModelText, startOffset: number, endOffset: number};
+type TextEdge = "startOffset" | "endOffset";
+
+function selectTextAndVoidNodes(node: ModelNode) {
+  if (ModelNode.isModelText(node)) {
+    return FilterResult.FILTER_ACCEPT;
+  }
+  else if (ModelNode.isModelElement(node) && VOID_NODES.includes(node.type)) {
+    return FilterResult.FILTER_ACCEPT;
+  }
+  else {
+    return FilterResult.FILTER_SKIP;
+  }
+}
+
+function buildTextNodeArrayWithOffsets(treeWalker: ModelTreeWalker): ModelTextWithOffsets[] {
+  const textNodes: ModelTextWithOffsets[] = [];
+  let startOffset = 0;
+  for (const node of treeWalker) {
+    if (ModelNode.isModelText(node)) {
+      const endOffset = startOffset + node.length;
+      textNodes.push({
+        node,
+        startOffset,
+        endOffset
+      });
+      startOffset = endOffset;
+    }
+    else {
+      // void nodes
+      startOffset = startOffset + 1;
+    }
+  }
+  return textNodes;
+}
+
+
+function findBestMatchingTextWithOffset(textNodes: ModelTextWithOffsets[], offset: number, preferredEdge: TextEdge): ModelTextWithOffsets {
+  const eligibleNodes = textNodes.filter((modelTextWithOffsets) => {
+    return modelTextWithOffsets.startOffset <= offset && modelTextWithOffsets.endOffset >= offset;
+  });
+  if (eligibleNodes.length == 1) {
+    return eligibleNodes[0];
+  }
+  else if (eligibleNodes.length > 1) {
+    // by definition this should include edges, prefer end
+    const eligibleNode = eligibleNodes.find((textWithOffsets) => textWithOffsets[preferredEdge] === offset);
+    if (! eligibleNode) {
+      throw `no eligble node found`;
+    }
+    return eligibleNode;
+  }
+  else {
+    throw `no eligible node found`;
```

Comment:
I'd prefer to see error instances here
If you cant find a suitable error type and dont want to make a new one, you can throw `new CustomError("no eligible node found")`

Reply:
I've created an IllegalArgumentError